docs: step by step - part 1 (#416)
jina-ai/finetuner: docs/2_step_by_step/2_2_experiment_and_run.md

# Two basic concepts: Experiment and Run

> Finetuner organize your training based on two concepts: **Experiment** and **Run**.

Reviewer: organizes

> An Experiment is defined as a general purpose machine learning task you're fine-tuning for.

Reviewer 2: ```suggestion
An Experiment defines the machine learning task you're fine-tuning for in general.
```
For me, a `general purpose ML task` would be something different, i.e., `classification` in general

> An Experiment is defined as a general purpose machine learning task you're fine-tuning for.

Reviewer 3: ``Experiment`` instead of Experiment?

Author: i think when using `Experiment`, it should be a piece of code. emphasis should use bold or it

> A Run is a piece of code that performs the Experiment with specific configurations.

Reviewer 3: The same of Run.

An Experiment contains a list of Runs,
each with different configurations.
For example:

> + Experiment: Fine-tune transformer on QuoraQA dataset.

Reviewer: a transformer

  - Run1: Use bert-based model
  - Run2: Use setence-transformer model.
+ Experiment: Fine-tune ResNet on WILD dataset.
  - Run1: Use ResNet18 with learning rate 0.01 and SGD optimizer.
  - Run2: Use ResNet50 with learning rate 0.01 and SGD optimizer.
  - Run3: Use ResNet50 with learning rate 0.0001 and Adam optimizer.

When you start the fine-tuning job, you can declare the `experiment_name` and `run_name` like this:

```python
import finetuner

finetuner.fit(
  ...,
  experiment_name='quora-qa-finetune',
  run_name='quora-qa-finetune-bert',
)
```

Please note that these two arguments are `Optional`.

> Finetuner will use the current working directory as default `experiment_name`,

Reviewer: as a default

> Finetuner will use the current working directory as a default `experiment_name`,

Reviewer 4: If not supplied, Finetuner will use the current ...
